src/sentry/utils/snuba.py

```diff
@@ -847,9 +848,31 @@ def raw_query(
     return bulk_raw_query([snuba_params], referrer=referrer, use_cache=use_cache)[0]
 
 
-SnubaQuery = Union[Request, MutableMapping[str, Any]]
 Translator = Callable[[Any], Any]
-RequestQueryBody = tuple[Request, Translator, Translator]
+
+
+@dataclasses.dataclass(frozen=True)
+class SnubaRequest:
+    request: Request
+    referrer: str | None  # this should use the referrer class
+    forward: Translator
+    reverse: Translator
+
+    def __post_init__(self) -> None:
+        self.validate()
+
+    def validate(self):
+        if self.referrer:
```

Comment:
Huh, this matches previous behaviour but it feels so weird that referrer isn't required

Reply:
Yeah, I thought this was weird too but let's leave the behaviour unchanged in this PR